core/script.runtime/src/kotlin/script/dependencies/resolvers.kt

```diff
@@ -47,18 +66,23 @@ interface DependenciesResolver : @Suppress("DEPRECATION") ScriptDependenciesReso
     }
 }
 
+// is File a could type to use here?
+// No way to get script name if file is not present, should add another property (val fileName: String)
 interface ScriptContents {
     val file: File?
     val annotations: Iterable<Annotation>
```

Comment:
`annotations` are actually lazily loaded (we need reflection to do this so it's relatively expensive). `Iterable` indicates that there is no container yet that holds all the instances. I think `Sequence` is better suited to express `lazy` and `no-container` ideas.
I like `filleAnnotations` btw. We just have to make sure that we have no plans to expand the meaning.